Master fails to build on 10.6: `Fl_Native_File_Chooser_MAC.mm: error: two consecutive '[' shall only introduce an attribute before '[' token`

**Description of the Bug Report**
Master branch fails to build on 10.6.8.

**To Reproduce**
Run the build, see it fail.

**Expected behavior**
It should build normally.

**Screenshots**
```
cd /opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/build/src && /opt/local/bin/g++-mp-14 -DFL_LIBRARY -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -Dfltk_shared_EXPORTS -I/opt/local/include -I/opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/build/src/.. -I/opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/fltk-61f132845ba66462435cd8247d87d45e76ff32c9/src/.. -pipe -Os -DNDEBUG -D_GLIBCXX_USE_CXX11_ABI=0 -Wno-deprecated -std=c++11 -arch ppc -mmacosx-version-min=10.6 -fPIC   -D_THREAD_SAFE -D_REENTRANT -MD -MT src/CMakeFiles/fltk-shared.dir/print_button.cxx.o -MF CMakeFiles/fltk-shared.dir/print_button.cxx.o.d -o CMakeFiles/fltk-shared.dir/print_button.cxx.o -c /opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/fltk-61f132845ba66462435cd8247d87d45e76ff32c9/src/print_button.cxx
/opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/fltk-61f132845ba66462435cd8247d87d45e76ff32c9/src/Fl_Native_File_Chooser_MAC.mm: In function '-[FLopenDelegate panel:shouldShowFilename:]':
/opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/fltk-61f132845ba66462435cd8247d87d45e76ff32c9/src/Fl_Native_File_Chooser_MAC.mm:501:77: error: two consecutive '[' shall only introduce an attribute before '[' token
  501 |   if ( fl_filename_match([filename fileSystemRepresentation], filter_pattern[ [nspopup indexOfSelectedItem] ]) ) return YES;
      |                                                                             ^
make[2]: *** [src/CMakeFiles/fltk.dir/Fl_Native_File_Chooser_MAC.mm.o] Error 1
. . .
/opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/fltk-61f132845ba66462435cd8247d87d45e76ff32c9/src/Fl_cocoa.mm: In function '-[FLView performDragOperation:]':
/opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/fltk-61f132845ba66462435cd8247d87d45e76ff32c9/src/Fl_cocoa.mm:2761:13: error: two consecutive '[' shall only introduce an attribute before '[' token
 2761 |     DragData[[data length]] = 0;
      |             ^
make[2]: *** [src/CMakeFiles/fltk-shared.dir/Fl_cocoa.mm.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/fltk-61f132845ba66462435cd8247d87d45e76ff32c9/src/Fl_Native_File_Chooser_MAC.mm: In function '-[FLopenDelegate panel:shouldShowFilename:]':
/opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/fltk-61f132845ba66462435cd8247d87d45e76ff32c9/src/Fl_Native_File_Chooser_MAC.mm:501:77: error: two consecutive '[' shall only introduce an attribute before '[' token
  501 |   if ( fl_filename_match([filename fileSystemRepresentation], filter_pattern[ [nspopup indexOfSelectedItem] ]) ) return YES;
      |                                                                             ^
make[2]: *** [src/CMakeFiles/fltk-shared.dir/Fl_Native_File_Chooser_MAC.mm.o] Error 1
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/build'
make[1]: *** [src/CMakeFiles/fltk-shared.dir/all] Error 2
```

**FLTK Version**
Please complete the following information and delete non-applicable lines:
 - If from Git, commit: 61f132845ba66462435cd8247d87d45e76ff32c9

**FLTK Configure / Build Options**
```
Executing:  cd "/opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/build" && /opt/local/bin/cmake -G "CodeBlocks - Unix Makefiles" -DCMAKE_BUILD_TYPE=MacPorts -DCMAKE_INSTALL_PREFIX="/opt/local" -DCMAKE_INSTALL_NAME_DIR="/opt/local/lib" -DCMAKE_SYSTEM_PREFIX_PATH="/opt/local;/usr" -DCMAKE_C_COMPILER="$CC" -DCMAKE_CXX_COMPILER="$CXX" -DCMAKE_OBJC_COMPILER="$CC" -DCMAKE_OBJCXX_COMPILER="$CXX" -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_POLICY_DEFAULT_CMP0060=NEW -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_COLOR_MAKEFILE=ON -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_MAKE_PROGRAM=/usr/bin/make -DCMAKE_MODULE_PATH="/opt/local/share/cmake/Modules" -DCMAKE_PREFIX_PATH="/opt/local/share/cmake/Modules" -DCMAKE_BUILD_WITH_INSTALL_RPATH:BOOL=ON -DCMAKE_INSTALL_RPATH="/opt/local/lib" -Wno-dev -DFLTK_BACKEND_WAYLAND=OFF -DFLTK_BACKEND_X11=OFF -DFLTK_BUILD_EXAMPLES=OFF -DFLTK_BUILD_SHARED_LIBS=ON -DFLTK_BUILD_TEST=OFF -DFLTK_USE_GL=ON -DFLTK_USE_SVG=OFF -DFLTK_USE_SYSTEM_LIBJPEG=ON -DFLTK_USE_SYSTEM_LIBPNG=ON -DFLTK_USE_SYSTEM_ZLIB=ON -DCMAKE_OSX_ARCHITECTURES="ppc" -DCMAKE_OSX_DEPLOYMENT_TARGET="10.6" -DCMAKE_OSX_SYSROOT="/" -DFLTK_ARCHFLAGS="" /opt/local/var/macports/build/_opt_local_ppcports_aqua_fltk/fltk-devel/work/fltk-61f132845ba66462435cd8247d87d45e76ff32c9
```

**Operating System / Platform:**
Please be as precise as possible, e.g. "Linux: Ubuntu 20.04"
 - OS: macOS
 - OS Version: 10.6.8
 - Processor: PowerPC

I don't know if I have 10.6.x anymore.
Oldest I have is 10.10.5, which fails in a few different spots:
```
[ 83%] Building CXX object fluid/CMakeFiles/fluid-lib.dir/app/Snap_Action.cxx.o
/usr/local/src/fltk-1.4.x.git/fluid/Project.cxx:220:44: error: use of undeclared identifier 'errno'
            project_path.c_str(), strerror(errno));
                                           ^
/usr/local/src/fltk-1.4.x.git/fluid/Project.cxx:241:44: error: use of undeclared identifier 'errno'
            app_work_dir.c_str(), strerror(errno));
                                           ^
/usr/local/src/fltk-1.4.x.git/fluid/Project.cxx:272:63: error: use of undeclared identifier 'errno'
      fprintf(stderr, "%s : %s\n", filename.c_str(), strerror(errno));
                                                              ^
/usr/local/src/fltk-1.4.x.git/fluid/Project.cxx:277:67: error: use of undeclared identifier 'errno'
      fl_message("Can't write %s: %s", filename.c_str(), strerror(errno));
                                                                  ^
* * *
[ 53%] Building CXX object fluid/CMakeFiles/fluid-lib.dir/app/templates.cxx.o
/usr/local/src/fltk-1.4.x.git/fluid/app/templates.cxx:100:57: error: use of undeclared identifier 'errno'
    fl_alert("Error writing %s: %s", filename, strerror(errno));
                                                        ^
/usr/local/src/fltk-1.4.x.git/fluid/app/templates.cxx:125:3: error: use of undeclared identifier 'errno'
  errno = 0;
  ^
/usr/local/src/fltk-1.4.x.git/fluid/app/templates.cxx:128:57: error: use of undeclared identifier 'errno'
    fl_alert("Error writing %s: %s", filename, strerror(errno));
                                                        ^
* * *
[ 53%] Building CXX object fluid/CMakeFiles/fluid-lib.dir/panels/codeview_panel.cxx.o
/usr/local/src/fltk-1.4.x.git/fluid/panels/codeview_panel.cxx:157:35: error: use of undeclared identifier 'malloc';
      did you mean 'valloc'?
      cv_source_filename = (char*)malloc(FL_PATH_MAX);
                                  ^~~~~~
                                  valloc
/usr/include/unistd.h:707:7: note: 'valloc' declared here
void    *valloc(size_t);                        
         ^
/usr/local/src/fltk-1.4.x.git/fluid/panels/codeview_panel.cxx:162:35: error: use of undeclared identifier 'malloc';
      did you mean 'valloc'?
      cv_header_filename = (char*)malloc(FL_PATH_MAX);
                                  ^~~~~~
                                  valloc
/usr/include/unistd.h:707:7: note: 'valloc' declared here
void    *valloc(size_t);                        
         ^
/usr/local/src/fltk-1.4.x.git/fluid/panels/codeview_panel.cxx:167:35: error: use of undeclared identifier 'malloc';
      did you mean 'valloc'?
      cv_design_filename = (char*)malloc(FL_PATH_MAX);
                                  ^~~~~~
                                  valloc
/usr/include/unistd.h:707:7: note: 'valloc' declared here
void    *valloc(size_t);                        
         ^
* * *
[ 52%] Building CXX object fluid/CMakeFiles/fluid-lib.dir/panels/widget_panel.cxx.o
/usr/local/src/fltk-1.4.x.git/fluid/panels/widget_panel.cxx:2102:7: error: use of undeclared identifier 'free'
      free(c);
      ^
/usr/local/src/fltk-1.4.x.git/fluid/panels/widget_panel.cxx:2126:7: error: use of undeclared identifier 'free'
      free(c);
      ^
* * *
/usr/local/src/fltk-1.4.x.git/fluid/widgets/Code_Editor.cxx:158:3: error: use of undeclared identifier 'free'
  free(text);
  ^
/usr/local/src/fltk-1.4.x.git/fluid/widgets/Code_Editor.cxx:159:3: error: use of undeclared identifier 'free'
  free(style);
  ^
/usr/local/src/fltk-1.4.x.git/fluid/widgets/Code_Editor.cxx:182:22: error: use of undeclared identifier 'malloc'; did
      you mean 'valloc'?
    char *b = (char*)malloc(n+2);
                     ^~~~~~
                     valloc
```


This patch seems to resolve all those 10.10.5 errors.
_(I don't have a way to test @MoAlyousef's fixes for 10.6.x)_

``` diff
diff --git a/fluid/Project.cxx b/fluid/Project.cxx
index 9eb5552..c7c517a 100644
--- a/fluid/Project.cxx
+++ b/fluid/Project.cxx
@@ -14,6 +14,7 @@
 //     https://www.fltk.org/bugs.php
 //
 
+#include <errno.h>	// strerror(errno)
 #include "Project.h"
 
 #include "io/String_Writer.h"
diff --git a/fluid/app/templates.cxx b/fluid/app/templates.cxx
index 802357e..630e3ec 100644
--- a/fluid/app/templates.cxx
+++ b/fluid/app/templates.cxx
@@ -13,7 +13,7 @@
 //
 //     https://www.fltk.org/bugs.php
 //
-
+#include <errno.h>
 #include "app/templates.h"
 
 #include "Fluid.h"
diff --git a/fluid/panels/codeview_panel.cxx b/fluid/panels/codeview_panel.cxx
index 745ad0a..579e4e6 100644
--- a/fluid/panels/codeview_panel.cxx
+++ b/fluid/panels/codeview_panel.cxx
@@ -16,6 +16,7 @@
 
 // generated by Fast Light User Interface Designer (fluid) version 1.0500
 
+#include <stdlib.h>	// malloc()
 #include "codeview_panel.h"
 #include "Fluid.h"
 #include "Project.h"
diff --git a/fluid/panels/widget_panel.cxx b/fluid/panels/widget_panel.cxx
index 61b480f..29d75f0 100644
--- a/fluid/panels/widget_panel.cxx
+++ b/fluid/panels/widget_panel.cxx
@@ -16,6 +16,7 @@
 
 // generated by Fast Light User Interface Designer (fluid) version 1.0500
 
+#include <stdlib.h>	// free()
 #include "widget_panel.h"
 #include "Fluid.h"
 #include "app/Snap_Action.h"
diff --git a/fluid/widgets/Code_Editor.cxx b/fluid/widgets/Code_Editor.cxx
index b4379c8..216ffab 100644
--- a/fluid/widgets/Code_Editor.cxx
+++ b/fluid/widgets/Code_Editor.cxx
@@ -20,6 +20,7 @@
 // Include necessary headers...
 //
 
+#include <stdlib.h>
 #include "widgets/Code_Editor.h"
 
 using namespace fld;
```

--- a/src/Fl_Native_File_Chooser_MAC.mm
+++ b/src/Fl_Native_File_Chooser_MAC.mm
@@ -498,7 +498,7 @@ static char *prepareMacFilter(int count, const char *filter, char **patterns) {
   BOOL isdir = NO;
   [[NSFileManager defaultManager] fileExistsAtPath:filename isDirectory:&isdir];
   if (isdir) return YES;
-  if ( fl_filename_match([filename fileSystemRepresentation], filter_pattern[ [nspopup indexOfSelectedItem] ]) ) return YES;
+  if ( fl_filename_match([filename fileSystemRepresentation], filter_pattern[([nspopup indexOfSelectedItem])]) ) return YES;
   return NO;
 }
 - (BOOL)panel:(id)sender shouldEnableURL:(NSURL *)url

--- a/src/Fl_cocoa.mm
+++ b/src/Fl_cocoa.mm
@@ -2758,7 +2758,7 @@ static FLTextInputContext* fltextinputcontext_instance = nil;
     NSData *data = [pboard dataForType:UTF8_pasteboard_type];
     DragData = (char *)malloc([data length] + 1);
     [data getBytes:DragData length:[data length]];
-    DragData[[data length]] = 0;
+    DragData[([data length])] = 0;
     Fl_Screen_Driver::convert_crlf(DragData, strlen(DragData));
   }
   else {

